src/synth/fluid_synth.c

```diff
@@ -1181,20 +1250,51 @@ fluid_synth_remove_default_mod(fluid_synth_t* synth, const fluid_mod_t* mod)
  * @param num MIDI controller number (0-127)
  * @param val MIDI controller value (0-127)
  * @return FLUID_OK on success, FLUID_FAILED otherwise
+ * Note: The function support MIDI CC global which will be send to
+ * all channels when the basic channel is in mode (OmniOn,Mono).
+ * The CC needs to be send on MIDI channel one below the 'basic Channel'
+ * of the receiver. 
  */
 int
 fluid_synth_cc(fluid_synth_t* synth, int chan, int num, int val)
 {
-  int result;
+  int result = FLUID_FAILED;
+  fluid_channel_t* channel;
   fluid_return_val_if_fail (num >= 0 && num <= 127, FLUID_FAILED);
   fluid_return_val_if_fail (val >= 0 && val <= 127, FLUID_FAILED);
   FLUID_API_ENTRY_CHAN(FLUID_FAILED);
 
-  if (synth->verbose)
-    FLUID_LOG(FLUID_INFO, "cc\t%d\t%d\t%d", chan, num, val);
-
-  fluid_channel_set_cc (synth->channel[chan], num, val);
-  result = fluid_synth_cc_LOCAL (synth, chan, num);
+  channel = synth->channel[chan];
+  if(channel->mode &  FLUID_CHANNEL_ENABLED)
+  {	/* chan is enabled */
+	  if (synth->verbose)
+		FLUID_LOG(FLUID_INFO, "cc\t%d\t%d\t%d", chan, num, val);
+	  fluid_channel_set_cc (channel, num, val);
+	  result = fluid_synth_cc_LOCAL (synth, chan, num);
+  }
+  else /* chan is disabled so it is a candidate for global channel */
+  {	/* looks for next basic channel */
+	int n_chan = synth->midi_channels; /* MIDI Channels number */
+	int basicchan ;
+	if (chan < n_chan-1) basicchan = chan + 1; /* next channel */
+	else  basicchan = 0; /* wrap to 0 */
+	channel = synth->channel[basicchan];
```

Comment:
1. Do we need this global channel thing at all?
1. Is it necessary to make this global channel check so complicated? Couldnt we just

```
 if (synth->channel[chan]->mode & FLUID_CHANNEL_BASIC && FLUID_CHANNEL_MODE_OMNIOFF_MONO)
  then
    /* send cc to all channels in this basic channel */
```

Reply:
>1 . Do we need this global channel thing at all?

Yes we need it, Global Channel is a MIDI specification for channel basic in mode 3 (Omni off - Mono) .

>Is it necessary to make this global channel check so complicated?

The electric guitar is an example of instrument that could be put in mode 3 . We need 6 MIDI channels each for one string. When playing the musician could do two actions at the same time: 
1) use the `vibrato lever` to produce a `global vibrato` on the six strings . (This could be done using CC pitch bend on `global channel`)
2) use a bending on string 1 only. (This could done using  CC pitch bend on the `individual string MIDI channel`) . 
We get string 1 with a different bend then the other strings.

That mean that `global channel` and ` individual channel` need to be independent each to others.

To differentiate `global channel` the implementation logic follows the MIDI specs (please see pdf chapter 3.4.12). Of course it is a bit complicated for fluidsynth, but it is simpler for the application making use of instrument in mode 3.

Reply:
> the implementation logic follows the MIDI specs (please see pdf chapter 3.4.12)

Ok.

> for channel basic in mode 3 (Omni off - Mono) .

In this case I'll correct the API comment.

Reply:
Ok, later, i will do the change stated above about `breath mode` comment and removing `basic channel settings`.